### DISCUSSION TITLE
How can I set different colors for bar in BarChart?

### DISCUSSION THREAD

**Discussion by klubkovkv** [mantinedev/mantine#5942] [Q&A]
I need to fill bars with different colors, pls help
![image](https://github.com/mantinedev/mantine/assets/67817874/a5eb2bcc-ebd8-4425-87e0-50ad57094d32)

**Comment by rtivital** [comment]
You can pass props down to recharts Bar component with `barProps`. The prop accepts either an object or a callback function. By assigning fill, you should be able to change bars colors.

**Reply by klubkovkv** [reply]
Yes, but then don't I need an barIndex or something to assign the needed color?

What comes to mind is something like this, but it won't work

```
const colors = ['#fff', '#ccc']

<BarChart barProps={({barIndex}) => ({fill: colors[barIndex]})} />
```

**Comment by Sajarin-M** [comment]
I'm doing it like this:
```jsx
<BarChart
  classNames={{ bar: 'custom-bar' }}
  series={[{ name: 'Bar Name', color: 'var(--bar-color)' }]}
/>
```

Then, in CSS:
```
.custom-bar .recharts-bar-rectangle:nth-of-type(1) {
  --bar-color: var(--mantine-color-red-6);
}
.custom-bar .recharts-bar-rectangle:nth-of-type(2) {
  --bar-color: var(--mantine-color-teal-6);
}
.custom-bar .recharts-bar-rectangle:nth-of-type(3) {
  --bar-color: var(--mantine-color-violet-6);
}
.custom-bar .recharts-bar-rectangle:nth-of-type(4) {
  --bar-color: var(--mantine-color-pink-6);
}
```

**Reply by klubkovkv** [reply]
Thanks.
how can I use this method to make the color in the tooltip match the color of the bar?

**Reply by Sajarin-M** [reply]
I don't have a solution for that yet. for now, I just hide the color indicator. 😔
